cli/azd/cmd/show.go

```diff
@@ -68,21 +66,17 @@ func newShowAction(
 	writer io.Writer,
 	azCli azcli.AzCli,
 	projectConfig *project.ProjectConfig,
-	resourceManager project.ResourceManager,
```

Comment:
I had to move away from injecting this (to creating it manually inside the body of the action) because to create it, you need an environment and the logic in the IoC container would cause us to prompt the user if they wanted to create the environment when it didn't exist.

I suspect I could have solved this by also registering a Lazy<project.ResourceManager> in the IoC container, and wire things up such that if the environment didn't exist we'd just have the Lazy return an error when we try to fetch the object, but I wasn't 100% sure if that's the pattern we'd want to use here.

Interested in feedback on what we should be doing for these sort of "opportunistic" dependencies.  We do want to be able to work even when an environment can't be found and return a subset of the complete information.

Interested in the team's thoughts here.